main.go

```diff
@@ -412,14 +415,25 @@ func mainErr(args []string) error {
 			return err
 		}
 		anyPrivate := false
-		for path := range listedPackages {
+		for path, pkg := range listedPackages {
 			if isPrivate(path) {
+				pkg.private = true
 				anyPrivate = true
-				break
 			}
 		}
 		if !anyPrivate {
-			return fmt.Errorf("GOPRIVATE %q does not match any packages to be built", envGoPrivate)
+			return fmt.Errorf("GOPRIVATE=%q does not match any packages to be built", envGoPrivate)
+		}
+		for path, pkg := range listedPackages {
+			if pkg.private == true {
```

Comment:
`if pkg.private {`

Reply:
:facepalm: thanks for fixing this. I don't mind such obvious fixes at all.